contrib/scrooge/src/python/pants/contrib/scrooge/tasks/thrift_linter_task.py

```diff
@@ -60,21 +78,36 @@ def _is_strict(self, target):
     # 1. the option --[no-]strict, but only if explicitly set.
     # 2. java_thrift_library target in BUILD file, thrift_linter_strict = False,
     # 3. options, --[no-]strict-default
-    options = self.get_options()
-    if options.get_rank('strict') > RankedValue.HARDCODED:
-      return self._to_bool(self.get_options().strict)
+    task_options = self.get_options()
+    subsystem_options = ScroogeLinter.global_instance().options
+
+    task_strict_configured = not task_options.is_default('strict')
+    subsystem_strict_configured = not subsystem_options.is_default('strict')
+    if task_strict_configured and subsystem_strict_configured:
+      raise ValueError(self.deprecation_template.format(option="strict"))
+    if subsystem_strict_configured:
+      return self._to_bool(subsystem_options.strict)
+    if task_strict_configured:
+      return self._to_bool(task_options.strict)
 
     if target.thrift_linter_strict is not None:
       return self._to_bool(target.thrift_linter_strict)
 
-    return self._to_bool(self.get_options().strict_default)
+    task_strict_default_configured = not task_options.is_default('strict_default')
+    subsystem_strict_default_configured = not subsystem_options.is_default('strict_default')
+    if task_strict_default_configured and subsystem_strict_default_configured:
+      raise ValueError(self.deprecation_template.format(option="strict_default"))
```

Comment:
nit: Rather than exposing the deprecation template for formatting, consider exposing a method that constructs the ValueError... `raise self.deprecated(option="strict_default")`.